linera-views/src/backends/lru_caching.rs

```diff
@@ -83,32 +128,149 @@ pub const DEFAULT_STORAGE_CACHE_CONFIG: StorageCacheConfig = StorageCacheConfig
     max_cache_size: 10000000,
     max_entry_size: 1000000,
     max_cache_entries: 1000,
+    max_cache_value_size: 10000000,
+    max_cache_find_keys_size: 10000000,
+    max_cache_find_key_values_size: 10000000,
 };
 
-enum CacheEntry {
+#[derive(Eq, Hash, PartialEq, Debug)]
+enum CacheKey {
+    Value(Vec<u8>),
+    FindKeys(Vec<u8>),
+    FindKeyValues(Vec<u8>),
+}
+
+enum ValueCacheEntry {
     DoesNotExist,
     Exists,
     Value(Vec<u8>),
 }
 
-impl CacheEntry {
+impl ValueCacheEntry {
     fn size(&self) -> usize {
         match self {
-            CacheEntry::Value(vec) => vec.len(),
+            ValueCacheEntry::Value(vec) => vec.len(),
             _ => 0,
         }
     }
 }
 
+struct FindKeysEntry(BTreeSet<Vec<u8>>);
```

Comment:
Maybe Merkle Patricia Trie could be better-suited for this since it's designed to store data that share a prefix.